pilot/pkg/serviceregistry/aggregate/maistra.go

```diff
@@ -0,0 +1,29 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregate
```

Comment:
I thought keeping extension `func`s in their own file would make it easier to carry them forward and minimize merging effort. Not entirely convinced about the file name though.

Reply:
Yes, but without it, we are not able to construct imported sets properly, as the types do not match and we are not able to get hostname et al for `importedService`. Unless we move a callee part to the same package we can't really unpeel aggregate registry entry [[1]](https://github.com/maistra/istio/blob/maistra-2.3/pilot/pkg/serviceregistry/aggregate/controller.go#L56) for this purpose.

I don't like this kind of argument, but... it's been done like that before except for the new file :) Do you have some alternative solution in mind?

https://github.com/maistra/istio/blob/46ba36429213416c90da254ced7c45e2a2db541b/pilot/pkg/serviceregistry/aggregate/controller.go#L427-L432

Reply:
The code in question assumes we can cast the `serviceregistry.Instance` 

https://github.com/maistra/istio/blob/46ba36429213416c90da254ced7c45e2a2db541b/pkg/servicemesh/federation/discovery/controller.go#L193-L199

I was thinking if we could use `registry.Provider() == provider.Federation` for that purpose, but I'm not sure if we only foresee providing one registry from the whole `Federation` feature moving forward.